src/lephare/data_retrieval.py

```diff
@@ -299,3 +303,55 @@ def _check_downloaded_files(file_names, completed_futures):
 
     print("All files downloaded successfully and are non-empty.")
     return True
+
+
+def config_to_required_files(keymap, base_url=None):
+    """Take a lephare config and return list of auxiliary files required for run.
+
+    For the sed lists these must be present in the auxiliary files directory. If
+    local full paths set the code will only retrieve opa, vega, and filters.
+
+    In addition to the specified files we also add opa and vega files. These
+    are always required.
+
+    We use the tau opacities by default.
+
+    Parameters
+    ==========
+    keymap : dict
+        The dictionary of config keys containing filters etc required.
+    base_url : str
+        Url to overwrite default base.
+    """
+    if base_url is None:
+        base_url = DEFAULT_BASE_DATA_URL
+    required_files = []
+    # We always need alloutputkeys.txt
+    required_files += ["alloutputkeys.txt"]
+    # Opacity always required
+    opa_list = ["opa/OPACITY.dat"] + [f"opa/tau{i:02d}.out" for i in np.arange(81)]
+    required_files += opa_list
+    # vega always required
+    vega_list = [
+        "vega/BD+17.sed",
+        "vega/BD+17o4708.sed",
+        "vega/SunLCB.sed",
+        "vega/VegaLCB.sed",
+        "vega/a0v.sed",
+        "vega/a0v_n.sed",
+    ]
+    required_files += vega_list
+    required_files += [f"filt/{f}" for f in keymap["FILTER_LIST"].value.split(",")]
+    # Get user specified sed lists
+    sed_keys = ["STAR_SED", "GAL_SED", "QSO_SED"]
+    for key in sed_keys:
+        try:
+            list_file = base_url + keymap[key].value
```

Comment:
If `keymap` is an standard python dictionary, I think that calling `foo['bar'].value` will throw an exception. You can just call `foo['bar']` I believe.